sympy/solvers/solveset.py

```diff
@@ -540,6 +540,7 @@ def _solveset(f, symbol, domain, _check=False):
     # _check controls whether the answer is checked or not
 
     from sympy.simplify.simplify import signsimp
+    from sympy.solvers.diophantine import diop_solve
```

Comment:
Yeah, actually I was little confused because `diophantine` returns set of tuples , where `diop_solve` returns solution tuple (easy to convert into FIniteSet). 
Now I think `diophantine` should be used.